aiohttp/web_urldispatcher.py

```diff
@@ -213,35 +271,32 @@ def handle(self, request):
 
     def __repr__(self):
         name = "'" + self.name + "' " if self.name is not None else ""
-        return "<StaticRoute {name}[{method}] {path} -> {directory!r}".format(
-            name=name, method=self.method, path=self._prefix,
-            directory=self._directory)
+        return "<StaticRoute {name}[GET] {path} -> {directory!r}".format(
+            name=name, path=self._prefix, directory=self._directory)
 
 
-class SystemRoute(Route):
+class View(metaclass=abc.ABCMeta):
 
-    def __init__(self, status, reason):
-        super().__init__(hdrs.METH_ANY, None, None)
-        self._status = status
-        self._reason = reason
-
-    def url(self, **kwargs):
-        raise RuntimeError(".url() is not allowed for SystemRoute")
+    def __init__(self, handler, name='', method=hdrs.METH_ANY):
+        self._name = name
+        self._method = upstr(method)
+        self._handler = handler
```

Comment:
Shouldn't here be that code now assuming that only View now is cares about handler type and coroutine nature:

```
         assert callable(handler), handler
         if (not asyncio.iscoroutinefunction(handler) and
                 not inspect.isgeneratorfunction(handler)):
             handler = asyncio.coroutine(handler)
```